service/user/user.go

```diff
@@ -41,6 +41,13 @@ func SignUpUser(user model.User) (model.UserResponse, string, int, error) {
 		return model.UserResponse{}, fmt.Sprintf("unable to create token: %v", err.Error()), 500, err
 	}
 
+	// implementaton code
+	estCount , CountErr := mongodb.CountFromCollection(user.ID)
```

Comment:
you don't need to call this here, on sign up the api_call_count is obviously zero

CountErr should be err, it is ok to override the previous err